pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/queries/ResourceBasedQueriesTest.java

```diff
@@ -260,6 +294,48 @@ private static Object[][] testResourceQueryTestCaseProviderBoth()
     return providerContent.toArray(new Object[][]{});
   }
 
+  @DataProvider
+  private static Object[][] testResourceQueryTestCaseProviderWithMetadata()
+      throws Exception {
+    Map<String, QueryTestCase> testCaseMap = getTestCases();
+    List<Object[]> providerContent = new ArrayList<>();
+    for (Map.Entry<String, QueryTestCase> testCaseEntry : testCaseMap.entrySet()) {
+      String testCaseName = testCaseEntry.getKey();
+      if (testCaseEntry.getValue()._ignored) {
+        continue;
+      }
+
+      List<QueryTestCase.Query> queryCases = testCaseEntry.getValue()._queries;
+      for (QueryTestCase.Query queryCase : queryCases) {
+        if (queryCase._ignored) {
+          continue;
+        }
+
+        if (queryCase._outputs != null) {
+          String sql = replaceTableName(testCaseName, queryCase._sql);
+          if (!sql.contains("basic_test")) {
+            continue;
+          }
```

Comment:
technically you can test this stats with numOfSegments on all test cases. also you can directly add a test to QueryRunnerTest instead of doing it here. but I will leave this up to you.. 

my thinking is: since you are not testing many different scenarios. it is best to just have 1 `@Test` section in QueryRunnerTest

Reply:
Can't do for all test cases since the metrics contain only segments queries from the servers but doesn't account for the segments that are pruned by the brokers. So in some cases the test fails.